adagrams/game.py

```diff
@@ -1,11 +1,159 @@
+        pool_list.remove(random_letter) #removing letter from pool_list
+        my_list.append(random_letter) #adding this letter to my_list
+
+    return my_list
+
+
+letter_bank = draw_letters() # retreiving my_list of 10 listter from draw_letter function
+
+word = "SANTAAA"
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    # getting user input as string 
+    upper_word = word.upper()
+
+    letter_bank_dict = {} #empty dictionary to store 
+    inputed_word_dict = {} #empty dictionary to store letter and frequency from user
+
+
+    # adding upper_word into inputed_word_dict with letter: count format
+    for letter in upper_word:
+        if letter not in inputed_word_dict:
+            inputed_word_dict[letter] = 1 #Initialize with one if letter added first time
+        else:
+            inputed_word_dict[letter] += 1 #Increment by 1 if letter added additionally
+    
+
+    # converting answer_list to a letter_bank_dict with a format of letter:count
+    for letter in letter_bank:
+        if letter not in letter_bank_dict:
+            letter_bank_dict[letter] = 1
+        else:
+            letter_bank_dict[letter] += 1
+
+
+    # comparing letter_bank_dict and inputed_word_dict 
+    for key, value in inputed_word_dict.items():
+        if key not in letter_bank_dict or value > letter_bank_dict[key]:  # checking if key from inputed_word_dict is in letter_bank_dict and checking if value of inputed_word_dict is greater than value in letter_bank_dict
+            return False
+        
+    return True
+
+
+x = uses_available_letters(word, letter_bank)
+
 
 def score_word(word):
-    pass
+    score_dict = {
+        'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1,
+        'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+        'D': 2, 'G': 2,
+        'B': 3, 'C': 3, 'M': 3, 'P': 3,
+        'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+        'K': 5,
+        'J': 8, 'X': 8,
+        'Q': 10, 'Z': 10
+    }
+    #getting user input as string 
+    upper_word = word.upper()
+    score = 0
+
+    # incrementing score with values from score_dict
+    for letter in upper_word:
+        if letter in score_dict:
+            score += score_dict[letter]
+    # If the length of the word is 7, 8, 9, or 10: score gets incremented by 8
+    if len(upper_word) in (7,8,9,10):
+        score += 8
+    
+    return score
+
+
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    
+    winning_score = 0
+    winning_word = None
+    word_score_dict = {}
+    first_ten_letter_word = None
+
+    # iterate over word_list to create score_list
+    for word in word_list:
+        running_score = score_word(word) # calculating score of word with score_word function
+        word_score_dict[word] = running_score
+
+    #keep track of first 10 letter word:
+        if len(word) == 10 and first_ten_letter_word is None:
+            first_ten_letter_word = word
```

Comment:
This logic here builds up `word_score_dict` which makes `get_highest_word_score` have 2 responsibilities (it also finds the highest score). Consider moving lines 128-134 into a helper function (maybe something called `build_score_dict`) and invoke that helper here.